### ISSUE TITLE
Encrypting C/C++ regression codes

### ISSUE THREAD

**Issue by amithmath** [bespoke-silicon-group/bsg_replicant#615]
Is there any way to encrypt C/C++  codes regression/example codes?

**Comment by drichmond** [comment]
Hi @amithmath, for what purpose? You want to encrypt the source, like Hardware IP vendors encrypt verilog?

**Comment by amithmath** [comment]
Yes I want to encrypt the source like encrypt verilog, @drichmond.  Based on examples presented in bsg_replicant and bsg_manycore, I  am developing some applications and I plan to release in AWS marketplace as an API without disclosing the source. The reason why I am not disclosing the source is, I plan to commercialise the applications.

**Comment by amithmath** [comment]
@drichmond, by any chance is encryption possible? I would say overall project purpose adding this feature  would be very helpful.

**Comment by taylor-bsg** [comment]
Hi, we don’t have any specific knowledge about this — let us know what you
figure out!
